extension/sourcesdk/nav_mesh.h

```diff
@@ -23,23 +24,60 @@ class IPathCost
 
 extern CNavMesh *TheNavMesh;
 
-class CNavMesh
+class CGameEventListener : public IGameEventListener2
```

Comment:
It's an implementation of the `IGameEventListener2` class, located here: https://github.com/ValveSoftware/source-sdk-2013/blob/0d8dceea4310fde5706b3ce1c70609d72a38efdf/mp/src/game/shared/GameEventListener.h

This only exists because `CNavMesh` inherits from this class and it contains the `m_bRegisteredForEvents` member variable. Because it's in the `shared` folder, it's part of the core of the Source Engine itself and not game-specific.

I'll move this class to its own file though.

Reply:
It is game specific. The source engine is divided into multiple branches, and CSGO branch happens to add an extra member to this class.